app/src/main/java/org/fossasia/pslab/communication/ScienceLab.java

```diff
@@ -872,11 +872,15 @@ public void captureHighResolutionTraces(String channel, int samples, double time
     }
 
     public HashMap<String, double[]> fetchTrace(int channelNumber) {
-        this.fetchChannel(channelNumber);
-        HashMap<String, double[]> retData = new HashMap<>();
-        retData.put("x", this.aChannels.get(channelNumber - 1).getXAxis());
-        retData.put("y", this.aChannels.get(channelNumber - 1).getYAxis());
-        return retData;
+        try {
+            this.fetchChannel(channelNumber);
+            HashMap<String, double[]> retData = new HashMap<>();
+            retData.put("x", this.aChannels.get(channelNumber - 1).getXAxis());
+            retData.put("y", this.aChannels.get(channelNumber - 1).getYAxis());
+            return retData;
+        } catch (IndexOutOfBoundsException e) {
```

Comment:
Well if the device is not connected there is no point in starting the background task rather than catching its exceptions.
You can prevent the call from Oscilloscope Activity by sciencelab.isConnected() and could you cross check that its updated when device is disconnected.
Thanks